spec/instance_title_fields_spec.rb

```diff
@@ -87,6 +88,119 @@
         expect(doc_hash[:title_245a_display]).to eq nil
       end
     end
+
+    context 'title_245_search' do
+      let(:sparql_conn) { double('sparql client') }
+      before(:each) do
+        allow(isd).to receive(:sparql).and_return(sparql_conn)
+      end
+      it 'is a rdfs:label String (not an Array - single valued Solr field)' do
+        solutions << RDF::Query::Solution.new(p: 'rdf-schema#label', o: 'foo')
+        expect(sparql_conn).to receive(:query).and_return(solutions)
+        expect(doc_hash[:title_245_search]).to be_a String
+        expect(doc_hash[:title_245_search]).to eq 'foo'
+      end
+      it 'takes first label if there are multiple values' do
+        solutions << RDF::Query::Solution.new(p: 'rdf-schema#label', o: 'foo')
+        solutions << RDF::Query::Solution.new(p: 'rdf-schema#label', o: 'bar')
+        expect(sparql_conn).to receive(:query).and_return(solutions)
+        expect(doc_hash[:title_245_search]).to eq 'foo'
+      end
+      it 'nil if there is no label' do
+        expect(sparql_conn).to receive(:query).and_return(solutions)
+        expect(doc_hash[:title_245_search]).to eq nil
+      end
+    end
+
+    context 'title_full_display' do
+      let(:sparql_conn) { double('sparql client') }
+      before(:each) do
+        allow(isd).to receive(:sparql).and_return(sparql_conn)
+      end
+      it 'is a resposibilityStatement String (not an Array - single valued Solr field)' do
```

Comment:
actually, it's a concatenation of title_display and resp statement ... so I would test maybe more like

```
context 'no responsibility statement'
     have resp statement be explicitly nil here
  it 'is the same as title_display String'
     best if we can possibly mock title_display value rather than the lookups in solutions, but that may not be easy
  it 'takes title_display value if there are multiple values for mainTitle'
  it 'nil if no title_display value'
end

context 'resp statement'
  it 'plain title_display value if resp statement exists but is empty'
  it 'resp statement comes from 'responsibilityStatement' property'  (question - is this a bf property?  if so, add prefix in description of test)
  it 'concatenates resp statement to title_display value with ' / ' separator'
  it 'takes first resp statement if there are multiple values for it'
  it 'removes trailing punct from responsiblity statement'
  it 'removes comb of trailing punct and spaces from resp statement'
end

context 'no title_display'
  it nil if no title_display or resp statement
  it 'resp statement without prefix separator if no title_display value
```